idunn/blocks/environment.py

```diff
@@ -66,3 +72,43 @@ def get_air_quality(geobbox):
     if not kuzzle_client.enabled:
         return None
     return kuzzle_client.fetch_air_quality(geobbox)
+
+
+class Weather(BaseBlock):
+    BLOCK_TYPE: ClassVar = "weather"
+
+    temperature: Optional[float] = None
+    icon: Optional[constr(regex="11d|09d|10d|13d|50d|01d|01n|02d|03d|04d|02n|03n|04n")]
+    _connection: ClassVar = None
+
+    @classmethod
+    def from_es(cls, place, lang):
+        if place.PLACE_TYPE != "admin":
+            return None
+        if place.get("zone_type") not in ("city", "city_district", "suburd"):
+            return None
+
+        coord = place.get_coord()
+        if not coord:
+            return None
+
+        try:
+            weather = get_local_weather(coord)
+        except Exception:
+            logger.warning("Failed to fetch weather for %s", place.get_id(), exc_info=True)
+            return None
+
+        if not weather:
+            return None
+
+        return cls(**weather)
+
+
+def get_local_weather(coord):
+    def inner(coord):
+        return weather_client.fetch_weather_places(coord)
+
+    if not weather_client.enabled:
+        return None
+    key = "{}_{}_{}".format(Weather.BLOCK_TYPE, coord["lat"], coord["lon"])
```

Comment:
also we should probably use approximate coord as key, otherwise the cached values will __probably__ never be accessed.

Reply:
It's based on cities so it shouldn't be an issue. But just in case, what precision would you use in here?